### ISSUE TITLE
False-Positive duplicate matrix config

### ISSUE THREAD

**Issue by DNKpp** [rhysd/actionlint#523]
Hello,
I'm using actionlint online checker from time to time. It's very helpful to me, thank you very much.

But recently I got the feeling, that there may be a bug in the duplicate checker.
```yml
- prefix: "Windows 2022"
  os: "windows-2022"
  compiler_name: "msvc"
  compiler_version: "v143"
  cmake_generator: "Visual Studio 17 2022"

- prefix: "Windows 2022"
  suffix: "/Gv"
  os: "windows-2022"
  compiler_name: "msvc"
  compiler_version: "v143"
  cmake_generator: "Visual Studio 17 2022"
  default_call_convention: "vectorcall"
```
It claims, that these two matrix entries would be a duplicate of each other, which they are not (at least from my knowledge).
```
duplicate value {"cmake_generator": "Visual Studio 17 2022", "compiler_name": "msvc", "compiler_version": "v143", "default_call_convention": "vectorcall", "os": "windows-2022", "prefix": "Windows 2022", "suffix": "/Gv"} is found in matrix "config". the same value is at line:182,col:13matrix
```
When I change the order, the error is gone, which leads me to the conclusion, that the duplicate check does only check whether any previous entry is a subset of the new element (not actually the same).

[Here](https://rhysd.github.io/actionlint/#eJztW3tz2rgW/z+f4ix0mmSKoSHpY7nD7hJwUm4JZICk3dt0GGMLcGNsryWH5Hbz3e+RZINtzLNJ28xdZlpiSzrS+em8dI7IwvRTddw7zxyOGNScsWmbOuzVmu9ddx/eO8S9JlB8WTwChX+9yu9koWZS5pl9nxEDfNsgHrARgWPHoQw6zoBNNI9Aw9SJTUkOLolHTceGg/xLPhg/ex1CQNN1Z+xq9p1pD2FgWjiiXlWbHbV30HuZZ7cMHA90XBloTA6TnxFjLi0VCpPJJN/nU+Ydb1hIjN3f2bG1MSlB3zctA54DI5TtOHZpB8D16Yh/A/Q9zdZHhJbgE4w1086BQW6I5bhjYjP4LPpaVs8jf/k4fL0xO1+cPuVds5EPf+T/oEYGpk2owIuzj2xzJu2BOfQ9jSFMFAaOxNPB/zzg5PLh+IDWkNgEexOlr1GijDXcjFu5Os+3qYJsgt/3beYrliYYF22Oz1yfUdkRQPP0kcmIznyPc/Ps61egjLg0L/spjiuWEzzSfKw/3N8HZATAvbFjrEUk0ntGQr+97d1IIVmHRrQ7JyKoiBEha1kgtta3cA6D9P2hwufLgf7iRfElaLYBr4/6JhM4a5YFOoJIg5EKSLFRxXjg+ILEdyeUPzoillXiTaPpO4S9BH9PHwGIPnIgU2lX39W7arV70VY75auMmPcqk4HffoNnp/Xuu4vjntq8nBt3fFFv1HpnrZoYVeM8rDGq+vFj71Jtd+qtZqdcfDnXfwqOOQCT7VLQ4LwNA88Zx0Q4kD4u24GwA/4lRc7KgWYY4BGLCMmb4noocD0sIn/pSJJbRtBSGEk0zYHY8Qgze0OTjfx+Hpdksx4nAuUy7EY1cReeP4eg24hoBr4fiE4RPnb3IzT//jvsjj1nNDmPu5FuU4n8hl1+tifePvsae32fg6uMgOcqs7/R/k/pRV4Kam25CevQi0nGlGD0LVIsHs4TSuxlS2hg3HLkpAVQpE5zKUB5UEL9nO2zUYK4Pm8DdWzmkJNPc2DDZ8lKJuSlddE9v+hm5uhFzFGUWhTqtWlFzVKUWAzmhdR2AkuqIIIKm3obmxADbWKawY/be2415fu8dCd5Z2ph8QVDSSdeWrdpY94ca0MSjpEbHu7A/DAXFcm8xd6wsEvo36S+3d8rS7oEwC2lR/1BfMq96osX0W64AZQhfppnYK9clMRso5MtUZHCtv3QnXC4h3ehRxlopqUMNNwW/NOiJHgddb38EyUmwebm9d+dVnNP7GQ+bSPTPex+1BrNlr851YiMx2hG4dqcalTa42TFZpUi6pEF3dLsYeSNAlJ8SpBpmLZ/G1UmxF7sM7YVLLOPxiTe6qA6ZGLRTbx9Juyx12iBuHjj2OFIR1l3CoZ97boFsbTSwa9JIhHRxTGi14IuAQrYK0mFL/8WGWGeT2INGtXs4PXaqDxBvqOasj3jP0Ic3j4ILG+fnDg8Kt9PVxzePAgsb56cODwq309XHF4/CCwJKo5t3fUGY5YiEE9AUn4UJOsKURaGuv6jMMKpSwdHyxHCPqvxOXpMTXoMpg8fgOnDp8Z08QGYLj6iJozpzQJV+GDahjOhPM+bXADHaSKblfnmJI98ipVM3hwcJeVjrF2TXnDucPC4mrk0qa9Z0GG+YTpw8CZY2qbLn3mJ05ufmrHoIIMMNN9iPV2zrB4KIs+HBRPgGdHx+PvNoXgMhqvcvFetb99MFE5Nb9F0nnhTJ4UZfO/QBdobZ6biuhapbhfEGgNLG9LexPGuNc/xbTwzZ5RGwXFZYeSMSd8jE/FgWTdjHosUMBiBpR18m28DKJb8I7OG8v4wNJIx3CZo/HHwZgbIByuneK7GRrmVnX9qQDYIVTbD6vUmWL1eHytyq1u+QeKJme7IpJy9vmmL4hN4hKLJoTzjr4V8A/E8x+OvPISGUOApd+LlY9sQzyYVD5NeNJETSoM/JUhcC/vFPjALlJAxxfZrIixLBwyHUHuXoUdwXcdjsmQh82txsxNP5cnc/VxskMJVUh5Fc6w6FWbTqyOiXzs+Q0LGbNU+5YU0TRc58oIe9Pnj5iiZjO8Q5ruyQiVMLayTV+eI4PbiQAR2jBseDMrxeQ3o38mKmKAXliVJUI6LECG3rmXqJrPuwDCpKPDwcZIYmvgB0WSZcES8GWsLkvrVs8p7tXdc6ai91nk3UaOYa7uHqwTiyz6KYjlDxeJloXJNPb443Wx0DeQCLtX2cQvXwB9O6g21/Kfa2ZTSWf2sXj0/75202lW11uNlgU630qxV2rVyZnEaO7PtPGqzctxQe5Va5byrtntdtdPtlFsnJ9vSq7aaJ/XTnvrxXG3Xz9Rmt9LoVSvd6rsiwoJfOEe92VVP2xW+Tw8900WzXm3VVISsHU730FPgblTfd9uVqvrgi0fRwbbNaa+syEkjIEzNlCQvqaIAeYx+MNloL1kVyvG6J/aPFEjXq8DZN8DdDzoCpI7abpv2Fw24I4r0ajQuz3rNyplaFh5qrZLPskmeTQkmJzmvdN+h4uyJ7ooSFKRm/ffnS3TV6tSyTEkU0PfJLMZaddRlBF68WLhf/IO217FuxOUP9LompT5uYHibJShPIzqFd6GfDx2+ojseKYgBtHDw5u3R4au5pTVqJ43K6cxyBs8pZbT5eISXvZYtfDszrdSCt6KY2v3zXC0vKsitKeYWP7/HxHxhXZJX98Vxf3fLov6P5zZ2TE9nNl5hlVcacNTmPGcxWNRsyoPkWOADzAHNdT3H9Uze5mKPgeONg9ggQsEcwKdPkFlcWM3w5ckbMBn4/Plf/JKJHTOD541KF53jWTlz+zqW6SLWmtRlkLaKOp6ND2On3oEZefiu4Qnu+elVJmVj42dnZPBqszgACXdTCc9b380pVzjlEM9Nxq9hYSNKpH5EhzmH9awJlU6RbC2s9K+paprFiMdPQDckzLoAz64os6yLuC32LSr5/Hmi54L0DvyCQ75NgxdRTiizjNaX6vCKpQZ6tzvLR+2mK1+10mjwyOgS4yEeHCbzcCSZu5TSULdvNMs0knuRhvuSJcYJ3+Jx72AN5c9CzYFmq4tH4rGDcsHvxBmOz486hRyesKUXn5jiAiEenhB/2Wr4Hr/RGj9DydP1Hs5N0amPnAk/QmGMT/mmVEuFc88ZetoYTlB2aOHUZAgQ/P7777/sz6vIx49xV/81fHMPBXyIQ71aBUTCA2T1jJ8OfXcjEef7L6KfFBGX53LehecitvXES/hVKDNwkrJc/WrzkoySvgYvkJKFBLS+uabFkGkD3NLGspgk5qQ4UGJYClAPFrUsw2p8WFzDACfHbzR6Cb6p45dgGwjkIogjUy8HO9JxNezLOseFOdJzxVZlQwsibx3zVJtnBlclw1OOZDm8TxXNtNTR6PB81bXtTHLADQcajHAY2qQxUBNdXf9OfOcXC4jmMmVIWLgY5S7htSXe693RUwxykzocFWh9CimL03z0I+EC0+2y4V4P0fFrhqHEdtw8fPs6haLvGtzXxaiFbSGM34xFiU/+7YAIMkmdCO5PVwzDI1RKzYVtiJ8RGNDRbJOZ/yXeCpPNE8LfaoY3PhB1Ks16t/4fnoKqtdVORyTNIq8vmjX1pN5Ua7xhyQE0CxXkE0boc7nzRB/lmY5Ppw53gkfn0FyQW52IsAbGuDZEh44jdChJP3HLkgz+p2B09AUFKjxsv/r16LCY2JDaLOHp6D2DyO8xzqQxfYTBF5dIvoLIujbZnhWGZ8s76oj6/L5F396X0hkqv1y4NdlVn7Af8EQUz3TPJYiTscg05XysUVOHE4yJCE9UzAA0SkFQxa+lmqvC8bAuw6PqrYRfhnHJA47Sgfz8y+Pg10bJBr4sYt/k59Vk+vOVGQTHgsRC9gfRH8hEgeA3c3UnCMior+toLTYPIyS7iiIZSWdH+fJqFYuzQ1oKh23fTvInfp5FH4tLXFi0SlPlqfDg4n2v1eydVOqNi7ZaihwL5nHhC0Rc+JdimN4iaKoLj6JpMIb6sbY+bax4J1LPSsB1gGduNMbwdIJg6tfENhaqn7BdS8f8LPozT2YuF8/T7ydnXe5m0rXtgbl9fPUR6rL1sv/RhxLGUCYvu/IfenDYhKvDYGy5QvhLBz1NjVij98pC3HLFekDUvpdmbbPkf7RKlB31a+ZpOlmuSJQZpdKsMyStViTKw9C3eKjMEZ4P+KJl89/KUDz8+WK9x9XMSP16uUKuRD8t/Eruw9OKNyMO88dx//NaiKkm6667hvqGveaw+/9Ts43dZ/QqyHI1fQiUv6tmbbTgn1cZ5F+P6C4rhuYy4kE3cujNwgc8Bw8sZwImBXptui4xcjAZERs0e60KJY7TXNcy+bg+kUUp22HyZt6AJ4unRbYsDKZH7+AmJAWT5ReqvRYsOXFOn7npoIPYi4UuenmprrxV4fMJWZPUG3lb2JAlt+4WWpQFG5jm66Jb+bS8/Pdk8tHsV4E/FqLro9/bpP0Pvi727Q==) is a link to the specific file.
